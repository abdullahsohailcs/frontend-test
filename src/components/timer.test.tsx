import { render } from '@testing-library/react';
import Timer from './timer';

describe('Timer Component', () => {
  beforeEach(() => {
    // Use fake timers for controlling time
    jest.useFakeTimers();
    // Mock the Date object to return a fixed time
    const fixedDate = new Date('2023-01-01T00:00:00.000Z');
    jest.spyOn(global, 'Date').mockImplementation(() => fixedDate as unknown as string);
  });

  afterEach(() => {
    // Clear timers and restore Date mock after each test
    jest.clearAllTimers();
    jest.restoreAllMocks();
  });

  it('matches the snapshot at a specific time', () => {
    // Render the Timer component
    const { asFragment, getByTestId } = render(<Timer />);
    
    // Advance timers by a fixed amount (to simulate the passage of time)
    jest.advanceTimersByTime(1000); // For example, advance by 1000ms (1 second)

    // Now capture the snapshot fragment
    const fragment = asFragment();

    // Match the snapshot
    expect(fragment).toMatchSnapshot();

    // Check the timer text content with expect.any
    expect(getByTestId('timer').textContent).toEqual(expect.any(String));
  });
});