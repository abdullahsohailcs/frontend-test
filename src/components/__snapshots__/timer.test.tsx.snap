// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer Component matches the snapshot at a specific time 1`] = `
<DocumentFragment>
  <div
    class="timer text-sm font-mono text-white px-4 py-1 rounded-lg flex items-center"
    data-testid="timer"
    style="background-color: rgb(51, 51, 51); box-shadow: 0px 4px 8px rgba(50, 50, 93, 0.2); max-width: 200px;"
  >
    05:00:00:000
  </div>
</DocumentFragment>
`;
